sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/export/IntervalMetricReader.java

```diff
@@ -62,11 +64,24 @@ public static IntervalMetricReaderBuilder builder() {
     this.exporter = new Exporter(internalState);
     this.scheduler =
         Executors.newScheduledThreadPool(1, new DaemonThreadFactory("IntervalMetricReader"));
-    this.scheduler.scheduleAtFixedRate(
-        exporter,
-        internalState.getExportIntervalMillis(),
-        internalState.getExportIntervalMillis(),
-        TimeUnit.MILLISECONDS);
+  }
+
+  /**
+   * Starts this {@link IntervalMetricReader} to report to the configured exporter.
+   *
+   * @return this for fluent usage along with the builder.
+   */
+  public IntervalMetricReader start() {
+    if (scheduledFuture != null) {
```

Comment:
Should we go ahead and use `AtomicReference` + CAS for the off chance of parallel start?

Reply:
I can do that, yes.

Reply:
actually...not sure how to do this, since you can't do a lazy populator for the atomic value. What I really want is an API that looks like:

`atomicReference.compareAndSet(null, ()-> schedule() )` but we don't have that API.  :(

I could do standard double-checked locking for this, or we could just punt and warn in the javadoc about this case. Thoughts?

Reply:
Or, since it's not a throughput-sensitive call, simple synchronization on a lock would probably be fine, rather than adding the complication of double-checked locking.